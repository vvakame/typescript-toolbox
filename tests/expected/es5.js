var Sample = (function () {
    function Sample() {
    }
    Object.defineProperty(Sample.prototype, "name", {
        get: function () {
            return this._name;
        },
        set: function (name) {
            this._name = name;
        },
        enumerable: true,
        configurable: true
    });
    return Sample;
})();
