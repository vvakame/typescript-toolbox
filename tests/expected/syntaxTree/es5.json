{
  "isDeclaration": false,
  "languageVersion": "EcmaScript3",
  "parseOptions": {
    "allowAutomaticSemicolonInsertion": true
  },
  "diagnostics": [
    {
      "start": 40,
      "length": 3,
      "diagnosticCode": "Accessors are only available when targeting ECMAScript 5 and higher."
    },
    {
      "start": 86,
      "length": 3,
      "diagnosticCode": "Accessors are only available when targeting ECMAScript 5 and higher."
    }
  ],
  "sourceUnit": {
    "kind": "SourceUnit",
    "fullWidth": 137,
    "isIncrementallyUnusable": true,
    "moduleElements": [
      {
        "kind": "ClassDeclaration",
        "fullWidth": 137,
        "modifiers": [],
        "classKeyword": {
          "kind": "ClassKeyword",
          "width": 5,
          "fullWidth": 6,
          "text": "class",
          "value": "class",
          "valueText": "class",
          "hasTrailingTrivia": true,
          "trailingTrivia": [
            {
              "kind": "WhitespaceTrivia",
              "text": " "
            }
          ]
        },
        "identifier": {
          "kind": "IdentifierName",
          "width": 6,
          "fullWidth": 7,
          "text": "Sample",
          "value": "Sample",
          "valueText": "Sample",
          "hasTrailingTrivia": true,
          "trailingTrivia": [
            {
              "kind": "WhitespaceTrivia",
              "text": " "
            }
          ]
        },
        "openBraceToken": {
          "kind": "OpenBraceToken",
          "width": 1,
          "fullWidth": 2,
          "text": "{",
          "value": "{",
          "valueText": "{",
          "hasTrailingTrivia": true,
          "hasTrailingNewLine": true,
          "trailingTrivia": [
            {
              "kind": "NewLineTrivia",
              "text": "\n"
            }
          ]
        },
        "classElements": [
          {
            "kind": "MemberVariableDeclaration",
            "fullWidth": 24,
            "modifiers": [
              {
                "kind": "PrivateKeyword",
                "width": 7,
                "fullWidth": 9,
                "text": "private",
                "value": "private",
                "valueText": "private",
                "hasLeadingTrivia": true,
                "hasTrailingTrivia": true,
                "leadingTrivia": [
                  {
                    "kind": "WhitespaceTrivia",
                    "text": "\t"
                  }
                ],
                "trailingTrivia": [
                  {
                    "kind": "WhitespaceTrivia",
                    "text": " "
                  }
                ]
              }
            ],
            "variableDeclarator": {
              "kind": "VariableDeclarator",
              "fullWidth": 13,
              "propertyName": {
                "kind": "IdentifierName",
                "width": 5,
                "text": "_name",
                "value": "_name",
                "valueText": "_name"
              },
              "typeAnnotation": {
                "kind": "TypeAnnotation",
                "fullWidth": 8,
                "colonToken": {
                  "kind": "ColonToken",
                  "width": 1,
                  "fullWidth": 2,
                  "text": ":",
                  "value": ":",
                  "valueText": ":",
                  "hasTrailingTrivia": true,
                  "trailingTrivia": [
                    {
                      "kind": "WhitespaceTrivia",
                      "text": " "
                    }
                  ]
                },
                "type": {
                  "kind": "StringKeyword",
                  "width": 6,
                  "text": "string",
                  "value": "string",
                  "valueText": "string"
                }
              }
            },
            "semicolonToken": {
              "kind": "SemicolonToken",
              "width": 1,
              "fullWidth": 2,
              "text": ";",
              "value": ";",
              "valueText": ";",
              "hasTrailingTrivia": true,
              "hasTrailingNewLine": true,
              "trailingTrivia": [
                {
                  "kind": "NewLineTrivia",
                  "text": "\n"
                }
              ]
            }
          },
          {
            "kind": "GetAccessor",
            "fullWidth": 46,
            "modifiers": [],
            "getKeyword": {
              "kind": "GetKeyword",
              "width": 3,
              "fullWidth": 5,
              "text": "get",
              "value": "get",
              "valueText": "get",
              "hasLeadingTrivia": true,
              "hasTrailingTrivia": true,
              "leadingTrivia": [
                {
                  "kind": "WhitespaceTrivia",
                  "text": "\t"
                }
              ],
              "trailingTrivia": [
                {
                  "kind": "WhitespaceTrivia",
                  "text": " "
                }
              ]
            },
            "propertyName": {
              "kind": "IdentifierName",
              "width": 4,
              "text": "name",
              "value": "name",
              "valueText": "name"
            },
            "parameterList": {
              "kind": "ParameterList",
              "fullWidth": 2,
              "openParenToken": {
                "kind": "OpenParenToken",
                "width": 1,
                "text": "(",
                "value": "(",
                "valueText": "("
              },
              "parameters": [],
              "closeParenToken": {
                "kind": "CloseParenToken",
                "width": 1,
                "text": ")",
                "value": ")",
                "valueText": ")"
              }
            },
            "typeAnnotation": {
              "kind": "TypeAnnotation",
              "fullWidth": 9,
              "colonToken": {
                "kind": "ColonToken",
                "width": 1,
                "fullWidth": 2,
                "text": ":",
                "value": ":",
                "valueText": ":",
                "hasTrailingTrivia": true,
                "trailingTrivia": [
                  {
                    "kind": "WhitespaceTrivia",
                    "text": " "
                  }
                ]
              },
              "type": {
                "kind": "StringKeyword",
                "width": 6,
                "fullWidth": 7,
                "text": "string",
                "value": "string",
                "valueText": "string",
                "hasTrailingTrivia": true,
                "trailingTrivia": [
                  {
                    "kind": "WhitespaceTrivia",
                    "text": " "
                  }
                ]
              }
            },
            "block": {
              "kind": "Block",
              "fullWidth": 26,
              "openBraceToken": {
                "kind": "OpenBraceToken",
                "width": 1,
                "fullWidth": 2,
                "text": "{",
                "value": "{",
                "valueText": "{",
                "hasTrailingTrivia": true,
                "hasTrailingNewLine": true,
                "trailingTrivia": [
                  {
                    "kind": "NewLineTrivia",
                    "text": "\n"
                  }
                ]
              },
              "statements": [
                {
                  "kind": "ReturnStatement",
                  "fullWidth": 21,
                  "returnKeyword": {
                    "kind": "ReturnKeyword",
                    "width": 6,
                    "fullWidth": 9,
                    "text": "return",
                    "value": "return",
                    "valueText": "return",
                    "hasLeadingTrivia": true,
                    "hasTrailingTrivia": true,
                    "leadingTrivia": [
                      {
                        "kind": "WhitespaceTrivia",
                        "text": "\t\t"
                      }
                    ],
                    "trailingTrivia": [
                      {
                        "kind": "WhitespaceTrivia",
                        "text": " "
                      }
                    ]
                  },
                  "expression": {
                    "kind": "MemberAccessExpression",
                    "fullWidth": 10,
                    "expression": {
                      "kind": "ThisKeyword",
                      "width": 4,
                      "text": "this",
                      "value": "this",
                      "valueText": "this"
                    },
                    "dotToken": {
                      "kind": "DotToken",
                      "width": 1,
                      "text": ".",
                      "value": ".",
                      "valueText": "."
                    },
                    "name": {
                      "kind": "IdentifierName",
                      "width": 5,
                      "text": "_name",
                      "value": "_name",
                      "valueText": "_name"
                    }
                  },
                  "semicolonToken": {
                    "kind": "SemicolonToken",
                    "width": 1,
                    "fullWidth": 2,
                    "text": ";",
                    "value": ";",
                    "valueText": ";",
                    "hasTrailingTrivia": true,
                    "hasTrailingNewLine": true,
                    "trailingTrivia": [
                      {
                        "kind": "NewLineTrivia",
                        "text": "\n"
                      }
                    ]
                  }
                }
              ],
              "closeBraceToken": {
                "kind": "CloseBraceToken",
                "width": 1,
                "fullWidth": 3,
                "text": "}",
                "value": "}",
                "valueText": "}",
                "hasLeadingTrivia": true,
                "hasTrailingTrivia": true,
                "hasTrailingNewLine": true,
                "leadingTrivia": [
                  {
                    "kind": "WhitespaceTrivia",
                    "text": "\t"
                  }
                ],
                "trailingTrivia": [
                  {
                    "kind": "NewLineTrivia",
                    "text": "\n"
                  }
                ]
              }
            }
          },
          {
            "kind": "SetAccessor",
            "fullWidth": 50,
            "modifiers": [],
            "setKeyword": {
              "kind": "SetKeyword",
              "width": 3,
              "fullWidth": 5,
              "text": "set",
              "value": "set",
              "valueText": "set",
              "hasLeadingTrivia": true,
              "hasTrailingTrivia": true,
              "leadingTrivia": [
                {
                  "kind": "WhitespaceTrivia",
                  "text": "\t"
                }
              ],
              "trailingTrivia": [
                {
                  "kind": "WhitespaceTrivia",
                  "text": " "
                }
              ]
            },
            "propertyName": {
              "kind": "IdentifierName",
              "width": 4,
              "text": "name",
              "value": "name",
              "valueText": "name"
            },
            "parameterList": {
              "kind": "ParameterList",
              "fullWidth": 15,
              "openParenToken": {
                "kind": "OpenParenToken",
                "width": 1,
                "text": "(",
                "value": "(",
                "valueText": "("
              },
              "parameters": [
                {
                  "kind": "Parameter",
                  "fullWidth": 12,
                  "modifiers": [],
                  "identifier": {
                    "kind": "IdentifierName",
                    "width": 4,
                    "text": "name",
                    "value": "name",
                    "valueText": "name"
                  },
                  "typeAnnotation": {
                    "kind": "TypeAnnotation",
                    "fullWidth": 8,
                    "colonToken": {
                      "kind": "ColonToken",
                      "width": 1,
                      "fullWidth": 2,
                      "text": ":",
                      "value": ":",
                      "valueText": ":",
                      "hasTrailingTrivia": true,
                      "trailingTrivia": [
                        {
                          "kind": "WhitespaceTrivia",
                          "text": " "
                        }
                      ]
                    },
                    "type": {
                      "kind": "StringKeyword",
                      "width": 6,
                      "text": "string",
                      "value": "string",
                      "valueText": "string"
                    }
                  }
                }
              ],
              "closeParenToken": {
                "kind": "CloseParenToken",
                "width": 1,
                "fullWidth": 2,
                "text": ")",
                "value": ")",
                "valueText": ")",
                "hasTrailingTrivia": true,
                "trailingTrivia": [
                  {
                    "kind": "WhitespaceTrivia",
                    "text": " "
                  }
                ]
              }
            },
            "block": {
              "kind": "Block",
              "fullWidth": 26,
              "openBraceToken": {
                "kind": "OpenBraceToken",
                "width": 1,
                "fullWidth": 2,
                "text": "{",
                "value": "{",
                "valueText": "{",
                "hasTrailingTrivia": true,
                "hasTrailingNewLine": true,
                "trailingTrivia": [
                  {
                    "kind": "NewLineTrivia",
                    "text": "\n"
                  }
                ]
              },
              "statements": [
                {
                  "kind": "ExpressionStatement",
                  "fullWidth": 21,
                  "expression": {
                    "kind": "AssignmentExpression",
                    "fullWidth": 19,
                    "left": {
                      "kind": "MemberAccessExpression",
                      "fullWidth": 13,
                      "expression": {
                        "kind": "ThisKeyword",
                        "width": 4,
                        "fullWidth": 6,
                        "text": "this",
                        "value": "this",
                        "valueText": "this",
                        "hasLeadingTrivia": true,
                        "leadingTrivia": [
                          {
                            "kind": "WhitespaceTrivia",
                            "text": "\t\t"
                          }
                        ]
                      },
                      "dotToken": {
                        "kind": "DotToken",
                        "width": 1,
                        "text": ".",
                        "value": ".",
                        "valueText": "."
                      },
                      "name": {
                        "kind": "IdentifierName",
                        "width": 5,
                        "fullWidth": 6,
                        "text": "_name",
                        "value": "_name",
                        "valueText": "_name",
                        "hasTrailingTrivia": true,
                        "trailingTrivia": [
                          {
                            "kind": "WhitespaceTrivia",
                            "text": " "
                          }
                        ]
                      }
                    },
                    "operatorToken": {
                      "kind": "EqualsToken",
                      "width": 1,
                      "fullWidth": 2,
                      "text": "=",
                      "value": "=",
                      "valueText": "=",
                      "hasTrailingTrivia": true,
                      "trailingTrivia": [
                        {
                          "kind": "WhitespaceTrivia",
                          "text": " "
                        }
                      ]
                    },
                    "right": {
                      "kind": "IdentifierName",
                      "width": 4,
                      "text": "name",
                      "value": "name",
                      "valueText": "name"
                    }
                  },
                  "semicolonToken": {
                    "kind": "SemicolonToken",
                    "width": 1,
                    "fullWidth": 2,
                    "text": ";",
                    "value": ";",
                    "valueText": ";",
                    "hasTrailingTrivia": true,
                    "hasTrailingNewLine": true,
                    "trailingTrivia": [
                      {
                        "kind": "NewLineTrivia",
                        "text": "\n"
                      }
                    ]
                  }
                }
              ],
              "closeBraceToken": {
                "kind": "CloseBraceToken",
                "width": 1,
                "fullWidth": 3,
                "text": "}",
                "value": "}",
                "valueText": "}",
                "hasLeadingTrivia": true,
                "hasTrailingTrivia": true,
                "hasTrailingNewLine": true,
                "leadingTrivia": [
                  {
                    "kind": "WhitespaceTrivia",
                    "text": "\t"
                  }
                ],
                "trailingTrivia": [
                  {
                    "kind": "NewLineTrivia",
                    "text": "\n"
                  }
                ]
              }
            }
          }
        ],
        "closeBraceToken": {
          "kind": "CloseBraceToken",
          "width": 1,
          "fullWidth": 2,
          "text": "}",
          "value": "}",
          "valueText": "}",
          "hasTrailingTrivia": true,
          "hasTrailingNewLine": true,
          "trailingTrivia": [
            {
              "kind": "NewLineTrivia",
              "text": "\n"
            }
          ]
        }
      }
    ],
    "endOfFileToken": {
      "kind": "EndOfFileToken",
      "width": 0,
      "text": ""
    }
  },
  "lineMap": {
    "lineStarts": [
      0,
      15,
      39,
      61,
      82,
      85,
      111,
      132,
      135,
      137
    ],
    "length": 137
  }
}